data.py

```diff
@@ -0,0 +1,101 @@
+            self.args["particles"] = list(map(self.string_processing, self.args["particles"]))
+
+        # Resolve relative paths to absolute paths
+        self.args["input_paths"] = list(map( (lambda x : Path(x).resolve()), self.args["input_paths"]))
+        self.args["output_paths"] = list(map( (lambda x : Path(x).resolve()), self.args["output_paths"]))
+
+    def string_processing(self, string):
+        return( string + "/fCoordinates/fCoordinates." )
+
+class CustomDataset(Dataset):
+    def __init__(self, manager):
+        self.manager = manager
+
+    def __len__(self):
+        file = uproot.open(self.manager.args["output_paths"][0])
+        len = file[self.manager.args["tree"]].arrays(self.manager.args["weights"], library="pd").shape[0]
+        file.close()
+        return len
+
+    def getitemhelper(self, idx):
+        for path in self.manager.args["input_paths"]:
+            file = uproot.open(path)
+            inputlist = []
+            for i in self.manager.args["particles"]:
+                inputlist.append( file[self.manager.args["tree"]].arrays([i+x for x in self.manager.args["4v_coords"]], library="pd", entry_start =idx, entry_stop =idx+1).values)
+            X = np.stack(inputlist, axis=-1)
+            file.close()
+        ## Stacking of multiple paths not implemented yet
+
+        for path in self.manager.args["output_paths"]:
+            file = uproot.open(path)
+            Y = file[self.manager.args["tree"]].arrays(self.manager.args["weights"], library="pd", entry_start =idx, entry_stop =idx+1).values
+            file.close()
+        ## Stacking of multiple paths not implemented yet
+
+        return X.squeeze(), Y.squeeze()
+
+    def getall(self):
+        for path in self.manager.args["input_paths"]:
+            file = uproot.open(path)
+            inputlist = []
+            for i in self.manager.args["particles"]:
+                inputlist.append( file[self.manager.args["tree"]].arrays([i+x for x in self.manager.args["4v_coords"]], library="pd").values)
+            X = np.stack(inputlist, axis=-1)
+            file.close()
+        ## Stacking of multiple paths not implemented yet
+
+        for path in self.manager.args["output_paths"]:
+            file = uproot.open(path)
+            Y = file[self.manager.args["tree"]].arrays(self.manager.args["weights"], library="pd").values
+            file.close()
+        ## Stacking of multiple paths not implemented yet
+
+        return X, Y
+
+    def __getitem__(self, idx):
+        return self.getitemhelper(idx)
+
+
+# path = "/data/deepmem_debug/tt_20evt.root"
+# path_weights = "/data/deepmem_debug/tt_20evt_weights.root"
+
+# file = uproot.open(path_weights)
+# a1 = file["t"].arrays(["bb_p4/fCoordinates/fCoordinates.fPt", "bb_p4/fCoordinates/fCoordinates.fEta", "bb_p4/fCoordinates/fCoordinates.fPhi", "bb_p4/fCoordinates/fCoordinates.fM"], library="pd").values.T
+# print(a1)
+# print(file["t"].show())
+
+# file.close()
+
+if(__name__ == "__main__"):
```

Comment:
```suggestion
if __name__ == "__main__":
```

The `()` aren't needed and don't do anything here.